libraries/Microsoft.Bot.Builder/SimplePropertyAccessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder
+{
+    public class SimplePropertyAccessor<T> : IPropertyAccessor<T>
+    {
+        private IPropertyContainer propertyContainer;
+        private T defaultValue;
+
+        public SimplePropertyAccessor(IPropertyContainer propertyContainer, string name, T defaultValue = default(T))
```

Comment:
Instead of taking a `T` here, I highly suggest taking a `Func<T>`. Require the person who is configuring the provider to take care of creating instances instead of owning that responsibility. Forget the `ICloneable`, forget the JSON pseudo-cloning, let the `Func<T>` own the semantics around how the `T` is created. If they want to hard code a constant in there, let them. Otherwise they will have only have to type a few more characters (`() =>`) and the framework is out of the business of providing these "new value" semantics itself.

Reply:
I've talked that over with Tom, and made much the same feedback. We're all on the same page there, that a Factory method makes more sense.